photutils/detection/findstars.py

```diff
@@ -376,6 +377,135 @@ def find_stars(self, data):
+        value indicates the corresponding element of ``data`` is masked.
+
+    border_width : bool, optional
+        The width in pixels to exclude around the border of the
+        ``data``.
+
+    npeaks : int, optional
+        The maximum number of peaks to return.  When the number of
+        detected peaks exceeds ``npeaks``, the peaks with the highest
+        peak intensities will be returned.
+
+    subpixel : bool, optional
+        If `True`, then a cutout of the specified ``box_size`` or
+        ``footprint`` will be taken centered on each peak and fit with a
+        2D Gaussian (plus a constant).  In this case, the fitted local
+        centroid and peak value (the Gaussian amplitude plus the
+        background constant) will also be returned in the output table.
+
+    error : array_like, optional
+        The 2D array of the 1-sigma errors of the input ``data``.
+        ``error`` is used only to weight the 2D Gaussian fit performed
+        when ``subpixel=True``.
+
+    wcs : `~astropy.wcs.WCS`
+        The WCS transformation to use to convert from pixel coordinates
+        to ICRS world coordinates.  If `None`, then the world
+        coordinates will not be returned in the output
+        `~astropy.table.Table`.
+    """
+
+    def __init__(self, threshold, box_size=3, footprint=None, mask=None,
+                 border_width=None, npeaks=np.inf, subpixel=True, error=None,
+                 wcs=None):
+        self.threshold = threshold
+        self.box_size = box_size
+        self.footprint = footprint
+        self.mask = mask
+        self.border_width = border_width
+        self.npeaks = npeaks
+        self.subpixel = subpixel
+        self.error = error
+        self.wcs = wcs
+
+    def __call__(self, data):
+        return self.find_stars(data)
+
+    def find_stars(self, data):
+        """
+        Parameters
+        ----------
+        data : array_like
+            The 2D array of the image.
+
+        Returns
+        -------
+        output : `~astropy.table.Table`
+            A table containing the x and y pixel location of the peaks and
+            their values.  If ``subpixel=True``, then the table will also
+            contain the local centroid and fitted peak value.
+        """
+
+        peaks = find_peaks(data, threshold=self.threshold,
+                           box_size=self.box_size, footprint=self.footprint,
+                           mask=self.mask, border_width=self.border_width,
+                           npeaks=self.npeaks, subpixel=self.subpixel,
+                           error=self.error, wcs=self.wcs)
+
+        if len(peaks) > 0:
+            if self.subpixel:
+                peaks['y_centroid'].name = 'ycentroid'
+                peaks['x_centroid'].name = 'xcentroid'
+            else:
+                peaks['y_peak'].name = 'ycentroid'
+                peaks['x_peak'].name = 'xcentroid'
+
+            peaks['id'] = np.arange(len(peaks)) + 1
+
```

Comment:
You mean if there are *no* peaks, right?  I would say not.  If the peak finder finds no peaks it might just be that the image really doesn't *have* any peaks that match the threshold, so you should have a warning when something is behaving correctly.

You might want to mention in the docstring, though, that `output` can be none (i.e. ``len(output)==0``), and that that's the case to check for to see if no peaks are found.